pages/index.vue

```diff
@@ -0,0 +1,124 @@
+<template>
+  <div class="flex flex-col gap-5">
+    <div class="w-[300px] flex flex-row gap-3 items-center">
+      <div>Category</div>
+      <USelectMenu class="w-[150px]" v-model="selectedName" :options="category" @change="handleCategorySelect" />
+    </div>
+    <div>
+      <div class="grid grid-cols-3 gap-5 " v-if="!isLoading">
+        <Card v-for="item in paginatedItems" class="hover:cursor-pointer" :items="item" />
+      </div>
+      <div v-else class="grid grid-cols-3 gap-5">
+        <UCard v-for="index in ITEMS_PER_PAGE" class="h-[400px] w-[100%]">
+          <USkeleton class="w-full h-[300px] bg-gray-200 mt-2 rounded" />
+          <USkeleton class="w-full h-6 bg-gray-200 mt-2 rounded" />
+        </UCard>
+      </div>
+    </div>
+    <div v-if="!filteredItems.length">
+      NO products to show
+    </div>
+    <div class="flex pb-2 w-full items-center justify-end">
+      <UPagination v-if="!filtered" v-model="currentPage" :page-count="ITEMS_PER_PAGE" :total="totalProducts" @click="handleChange" />
+      <UPagination v-if="filtered" v-model="currentPage" :page-count="ITEMS_PER_PAGE" :total="filteredItems.length" @click="handleChange" />
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+interface Item {
+  id: number;
+  name: string;
+  category: {
+    id: number;
+    name: string;
+  }
+}
+
+const currentPage = ref<number>(1);
+const ITEMS_PER_PAGE = 6;
+const items = ref<Item[]>([]);
+const paginatedItems = ref<Item[]>([]);
+const totalProducts = ref<Number>(0)
+const category = ref<string[]>([]);
+const categories = ref<Object>()
```

Comment:
```suggestion
const categories = ref<Record<string,unknown >>({})
```

Reply:
@10shubham01  I have resolved the comment